app/main/login.py

```diff
@@ -102,19 +102,19 @@ def change_password(token):
     except SignatureExpired:
         current_app.logger.info("Password reset attempt with expired token.")
         flash('token_expired', 'error')
-        return redirect(url_for('.forgotten_password'))
+        return redirect(url_for('.request_password_reset'))
     except BadSignature as e:
         current_app.logger.info("Error changing password: %s", e)
         flash('token_invalid', 'error')
-        return redirect(url_for('.forgotten_password'))
+        return redirect(url_for('.request_password_reset'))
     template_data = main.config['BASE_TEMPLATE_DATA']
-    return render_template("auth/change-password.html",
+    return render_template("auth/reset-password.html",
                            form=ChangePasswordForm(email_address=email_address,
                                                    user_id=user_id,),
                            **template_data), 200
 
 
-@main.route('/change-password', methods=["POST"])
+@main.route('/update-password', methods=["POST"])
```

Comment:
I'm not sure about having a different URL here from the /reset_password/ - nice to have GET and POST at the same URL to make clear that the form from the GET request POSTs to here.